tests/unit/src/test/scala/tests/NewFilesLspSuite.scala

```diff
@@ -2,17 +2,36 @@ package tests
 
 import java.nio.file.Files
 import scala.meta.internal.metals.ServerCommands
-import scala.meta.internal.metals.MetalsNewScalaFileParams
 import scala.meta.internal.metals.MetalsEnrichments._
 import scala.meta.internal.metals.RecursivelyDelete
+import scala.meta.internal.metals.MetalsServerConfig
+import scala.meta.internal.metals.Messages.NewScalaFile
+import scala.meta.internal.metals.MetalsInputBoxResult
 
 class NewFilesLspSuite extends BaseLspSuite("new-files") {
+  override def serverConfig: MetalsServerConfig =
+    super.serverConfig.copy(isInputBoxEnabled = true)
+
   test("new-worksheet") {
     cleanCompileCache("a")
     RecursivelyDelete(workspace.resolve("a"))
     Files.createDirectories(
       workspace.resolve("a/src/main/scala/").toNIO
     )
+    client.showMessageRequestHandler = { params =>
```

Comment:
Maybe a utility closure:
```
def createMessageRequestHander(name : String) : MessageRequestParams => Option[MessageRequestResult] =
{ params =>
      if (NewScalaFile.isSelectTheKindOfFile(params)) {
        params.getActions().asScala.find(_.getTitle() == name)
      } else {
        None
      }
    }
```

then you can just use:
`client.showMessageRequestHandler = createMessageRequestHander("worksheet") `

Reply:
Or even better create a check method:
```
def check(picked : String, expectedOutput : String, name : String = "Foo", shouldCheckForName : Boolean)
```
then you can just have:
```
check(
  "object",
 """| object Foo {
     |}
     |""".stripMargin
```

createMessageRequestHande might not be needed then.

Reply:
not sure, if I got you correctly, but I've done something there - please, check.